disnake/state.py

```diff
@@ -1222,6 +1252,23 @@ def parse_guild_member_update(self, data) -> None:
                 self.dispatch("user_update", user_update[0], user_update[1])
 
             self.dispatch("member_update", old_member, member)
+
+            if member.id != self.self_id:
+                return
+            # figure out which threads are no longer available
+            for thread in guild.threads:
+                # calculate old and new permissions
+                old_perm_value = thread.permissions_for(old_member).value
+                new_perm_value = thread.permissions_for(member).value
+                # get bits that were flipped from 1 to 0
+                diff_perm_value = old_perm_value ^ new_perm_value & old_perm_value
+                if (
+                    diff_perm_value
+                    and Permissions(diff_perm_value).view_channel
+                    and member.id in thread._members
+                ):
+                    self.dispatch("thread_remove", thread)
```

Comment:
Is this technically accurate? The thread still exists, and the bot *is* still a member, even if the bot cannot see it.

Reply:
It's not. However, there's no other way to know if you're removed from a private thread or not. If we don't dispatch thread_remove here and someone removes the bot afterwards, we won't dispatch it at all.